src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -62,6 +64,274 @@ static ShardPlacement * TupleToShardPlacement(TupleDesc tupleDesc,
+
+		totalRelationSize = list_length(workerNodeList) * currentRelationSize;
+		return totalRelationSize;
+	}
+
+	foreach(workerNodeCell, workerNodeList)
+	{
+		PGresult *queryResult = NULL;
+		StringInfo tableSizeQuery = NULL;
+		WorkerNode *workerNode = (WorkerNode *) lfirst(workerNodeCell);
+		char *workerNodeName = workerNode->workerName;
+		uint32 workerNodePort = workerNode->workerPort;
+		char *tableSizeString;
+		int executeCommand = 0;
+		uint32 connectionFlag = 0;
+		uint64 currentRelationSize = 0;
+
+		List *intervalsOnNode = GetShardIntervalsOnNode(workerNode, relationId);
+
+		tableSizeQuery = GenerateSizeQueryOnMultiplePlacements(relationId,
+															   intervalsOnNode,
+															   sizeQuery);
+
+		connection = GetNodeConnection(connectionFlag, workerNodeName, workerNodePort);
+
+		executeCommand = ExecuteOptionalRemoteCommand(connection, tableSizeQuery->data,
+													  &queryResult);
+
+		/* If there is a connection error close the connection and quit */
+		if (executeCommand != 0)
+		{
+			ReportConnectionError(connection, ERROR);
+		}
+
+		tableSizeString = PQgetvalue(queryResult, 0, 0);
+		currentRelationSize = atol(tableSizeString);
+		totalRelationSize += currentRelationSize;
+
+		PQclear(queryResult);
+
+		ForgetResults(connection);
+	}
+
+	return totalRelationSize;
+}
+
+
+/*
+ * citus_total_relation_size accepts a table name and
+ * returns a distributed table and its indexes' total relation size.
+ */
+Datum
+citus_total_relation_size(PG_FUNCTION_ARGS)
+{
+	Oid relationId = PG_GETARG_OID(0);
+	uint64 totalRelationSize = 0;
+
+	totalRelationSize = SupportFunctionForCitusSize(relationId,
+													INNER_SHARD_TOTAL_RELATION_SIZE_QUERY);
+
+	PG_RETURN_INT64(totalRelationSize);
+}
+
+
+/*
+ * citus_table_size accepts a table name and
+ * returns a distributed table's total relation size.
+ */
+Datum
+citus_table_size(PG_FUNCTION_ARGS)
+{
+	Oid relationId = PG_GETARG_OID(0);
+	uint64 tableSize = 0;
+
+	tableSize = SupportFunctionForCitusSize(relationId, INNER_SHARD_TABLE_SIZE_QUERY);
+
+	PG_RETURN_INT64(tableSize);
+}
+
+
+/*
+ * citus_relation_size accept a table name and
+ * returns a relation's 'main' fork's size.
+ */
+Datum
+citus_relation_size(PG_FUNCTION_ARGS)
+{
+	Oid relationId = PG_GETARG_OID(0);
+	uint64 relationSize = 0;
+
+	relationSize = SupportFunctionForCitusSize(relationId,
+											   INNER_SHARD_RELATION_SIZE_QUERY);
+
+	PG_RETURN_INT64(relationSize);
+}
+
+
+/*
+ * ErrorIfNotSuitableToGetSize determines whether the table is suitable to find its' size
+ * with internal functions.
+ */
+void
+ErrorIfNotSuitableToGetSize(Oid relationId)
+{
+	char *relationName = get_rel_name(relationId);
+
+	if (!IsDistributedTable(relationId))
+	{
+		ereport(ERROR, (errcode(ERRCODE_INVALID_TABLE_DEFINITION),
+						errmsg("cannot get the size"),
+						errdetail("relation %s is not distributed",
+								  relationName)));
+	}
+
+	if (TableReplicationModel(relationId) != REPLICATION_MODEL_STREAMING &&
```

Comment:
Requiring REPLICATION_MODEL_STREAMING is a bit invasive, since this is not the default. There are currently 0 Citus users with streaming tables. Could we express this restriction differently (no shards have replicas) or remove it?